File: additive_primes.py
# https://rosettacode.org/wiki/Additive_primes

def check_prime(n: int) -> bool:
    if n == 1: return False

    for i in range(2, n // 2 + 1):
        if n % i == 0:
            return False
    return True

def get_digit_sum(n: int) -> int:
    total = 0
    while n > 0:
        total += n % 10
        n //= 10
    return total

def get_additive_primes(limit: int = 500) -> list[int]:
    return [i for i in range(1, limit + 1) if check_prime(i) and check_prime(get_digit_sum(i))]

def main():
    nums = get_additive_primes()
    print(*nums)

    print(f"There are {len(nums)} Additive Primes under 500")

if __name__ == '__main__':
    main()

File: nim/additive_primes.nim
proc check_prime(n: int): bool =
  if n == 1: return false

  for i in 2..n div 2:
    if n mod i == 0: return false
  return true

proc digit_sum(n: int): int =
  var
    total = 0
    v = n
  while v > 0:
    total += v mod 10
    v = v div 10
  return total

proc additive_primes(limit: int = 500): seq[int] =
  var items: seq[int] = @[]
  for i in 1..(limit + 1):
    if check_prime(i) and check_prime(digit_sum(i)):
      items.add(i)
  return items

proc main() =
  var nums = additive_primes()
  for i in nums:
    stdout.write i, " "
  echo ""
  echo "Found ", nums.len, " additive primes under 500"

main()

